### ISSUE TITLE
Unexpected JArray(List()) extracted from JArray(List()) 

### ISSUE THREAD

**Issue by vitalscientist** [json4s/json4s#562]
I don't quite understand this behavior.

```
import org.json4s._
import org.json4s.native.JsonMethods._
import org.json4s.JsonDSL._

scala> val x = parse("{\"a\" : []}")
scala> x \ "a"
res0: org.json4s.JValue = JArray(List())  <- expected
scala> x \ "c"
res2: org.json4s.JValue = JNothing  <- expected as there is no "c", it should be JNothing
scala> x \ "a" \ "c"
res1: org.json4s.JValue = JArray(List()) <- unexpected as I expect JNothing as in json4s 3.2.11
```
What is the rationale of this change?  Any help is appreciate

### json4s version
"org.json4s" %% "json4s-native" % "3.5.2",
"org.json4s" %% "json4s-jackson" % "3.5.2",

### scala version
scalaVersion := "2.11.8"

### jdk version
java version "1.8.0_112"

**Comment by magnolia-k** [comment]
@vitalscientist 

Perhaps this commit is affecting. 

https://github.com/json4s/json4s/commit/a5d306728f8da40f2e0fec97034641b5e94382b8

This commit changes the behavior so that the JArray always returns when targeting the JArray.
In the previous code, JNothing was returned.

The intent of this commit is to return a JArray if there is only one element, but it will now return a JArray even if there are no elements.

**Comment by magnolia-k** [comment]
If it is such code, it will return to the original behavior.

```scala
--- a/core/src/main/scala/org/json4s/MonadicJValue.scala
+++ b/core/src/main/scala/org/json4s/MonadicJValue.scala
@@ -35,6 +35,7 @@ class MonadicJValue(jv: JValue) {
    * </pre>
    */
   def \(nameToFind: String): JValue = jv match {
+    case JArray(Nil) => JNothing
     case JArray(xs) => JArray(findDirectByName(xs, nameToFind))
     case _ =>
       findDirectByName(List(jv), nameToFind) match {
```

**Comment by magnolia-k** [comment]
https://github.com/json4s/json4s/pull/604

I changed the code a little

**Comment by seratch** [comment]
Closed by #604 thanks @magnolia-k
